qiskit_ionq/ionq_job.py

```diff
@@ -180,18 +176,22 @@ def __init__(
 
         if passed_args is not None:
             self.extra_query_params = passed_args.pop("extra_query_params", {})
-            self.extra_metadata = {
-                **passed_args.pop("extra_metadata", {}),
-                **(circuit.metadata or {}),
-            }
+            self.extra_metadata = passed_args.pop("extra_metadata", {})
             self._passed_args = passed_args
         else:
             self.extra_query_params = {}
             self.extra_metadata = {}
             self._passed_args = {"shots": 1024, "sampler_seed": None}
 
+        # Handle both single and list of circuits
         if circuit is not None:
-            self.circuit = circuit
+            if isinstance(circuit, (list, tuple)):
+                self.circuit = circuit
+                for circ in circuit:
+                    self.extra_metadata.update(circ.metadata or {})
```

Comment:
Do we still want to combine all metadata at the top level like this?

Reply:
ah we probably shouldnt to avoid going over the character limit! ill remove this too